geopandas/io/sql.py

```diff
@@ -410,6 +409,10 @@ def _write_postgis(
     # Convert geometries to EWKB
     gdf = _convert_to_ewkb(gdf, geom_name, srid)
 
+    # Replace SRID from -1 (geopandas pattern) to 0 (PostGIS pattern)
```

Comment:
just having a quick scan, it seems like we might be able to just update `_get_srid_from_crs` above to return an srid of 0 directly for none, instead of -1, and then we wouldn't need to do this string replacement (I may be wrong about that though, very quick look).

(this would also remove the need for the extra `and (srid != 0)` below I think)

Reply:
Thanks again, @m-richards ! 

You are right. I didn't check earlier if the function is called by another one. It is called exclusively by `_write_postgis`.

I made this change it on `_write_postgis` because I'm not sure if "-1" is something like a Geopandas null CRS pattern or it is not really defined (and I don't know where to get this info). If it's a pattern, maybe the correct way is to keep the changes on postgis method only, and let `_get_srid_from_crs` as a generic function.

But else, let me know and I made the changes on `_get_srid_from_crs`.

Reply:
Looking in a bit more detail, `-1` is the unknown srid convention of geoalchemy 2. So we shouldn't change `_get_srid_from_crs`. However, we should convert `srid` from -1 to 0 before passing it to `_convert_to_ewkb` - so that we don't have to do the string operation apply you have below.

Reply:
Nice. As I thought, make sense to keep `-1`.  Looking a little bit more, replacing `srid` before call `_convert_to_ewkb` is really the better solution!